<a href="https://colab.research.google.com/github/JuanARojasA/Ciencia_Datos/blob/master/RedesNeuronales.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import pandas as pd
import sklearn
from sklearn import datasets
from sklearn.preprocessing import StandardScaler
from sklearn.neural_network import MLPClassifier
from sklearn.metrics import classification_report

In [15]:
df = pd.read_csv('https://raw.githubusercontent.com/JuanARojasA/Ciencia_Datos/master/Taller7/semeion.csv', header=None)
df = df.sample(frac=1)

x_train = df.iloc[0:1116,0:256]
y_train = df.iloc[0:1116,256]
x_test = df.iloc[1116:1593,0:256]
y_test = df.iloc[1116:1593,256]

print("Printing some sample data from the iris dataset")
for training_sample in list(zip(x_train,y_train))[:5]:
    print(training_sample)

Printing some sample data from the iris dataset
(0, 8)
(1, 0)
(2, 3)
(3, 5)
(4, 8)


In [44]:
iterations=1000 
hidden_layers=[16,16,16]
 
mlp = MLPClassifier(hidden_layer_sizes=(hidden_layers), max_iter=iterations) 
mlp.fit(x_train, y_train)

MLPClassifier(activation='relu', alpha=0.0001, batch_size='auto', beta_1=0.9,
              beta_2=0.999, early_stopping=False, epsilon=1e-08,
              hidden_layer_sizes=[16, 16, 16], learning_rate='constant',
              learning_rate_init=0.001, max_fun=15000, max_iter=1000,
              momentum=0.9, n_iter_no_change=10, nesterovs_momentum=True,
              power_t=0.5, random_state=None, shuffle=True, solver='adam',
              tol=0.0001, validation_fraction=0.1, verbose=False,
              warm_start=False)

In [45]:
predicted = mlp.predict(x_test)  
 
print("Evaluation: considering the confusion matrix")
from sklearn.metrics import confusion_matrix
print(confusion_matrix(y_test,predicted))  

Evaluation: considering the confusion matrix
[[52  0  0  0  2  3  0  0  1  0]
 [ 0 42  1  0  1  0  0  1  1  0]
 [ 0  1 50  0  1  1  1  0  2  0]
 [ 0  0  0 37  0  0  0  2  1  5]
 [ 0  3  0  0 41  0  0  0  0  0]
 [ 0  0  0  1  0 36  1  1  1  0]
 [ 1  0  0  0  1  1 47  0  0  0]
 [ 1  0  1  2  1  0  0 43  0  0]
 [ 0  0  1  0  0  0  0  0 42  1]
 [ 0  0  0  0  1  2  0  0  0 43]]


In [46]:
print("Evaluation report:")
print(classification_report(y_test,predicted)) 

Evaluation report:
              precision    recall  f1-score   support

           0       0.96      0.90      0.93        58
           1       0.91      0.91      0.91        46
           2       0.94      0.89      0.92        56
           3       0.93      0.82      0.87        45
           4       0.85      0.93      0.89        44
           5       0.84      0.90      0.87        40
           6       0.96      0.94      0.95        50
           7       0.91      0.90      0.91        48
           8       0.88      0.95      0.91        44
           9       0.88      0.93      0.91        46

    accuracy                           0.91       477
   macro avg       0.91      0.91      0.91       477
weighted avg       0.91      0.91      0.91       477



El sistema para la clase 0, obtuvo una precisión del 96%, lo cual indíca cuales datos de esta clase fueron clasificados correctamente, con respecto a todos aquellos que se clasificaron como clase 0.

La recuperación del sistema (Recall), para la clase 1 fue del 91%, lo cual índica cuales datos se clasificaron correctamente, con respecto a la cantidad total de datos que pertenecían a esta clase.

La exactitu del sistema fue del 91% en promedio, lo cual índica que de la totalidad de la muestra se clasificaron correctamente esa cantidad.

El support índica la cantidad de datos de la clase.

El f1-Score es un balance entre la recuperación y la precisión del sistema. Esta medida es una mejora de la exactitud, ya que permite tener una mejor metrica al tener en cuenta la cantidad de negativos.

Juan Andrés Rojas Álvarez - 20141020031 \\
Cristian David Santoyo Parra - 20141020077